sdk/iothub/Azure.ResourceManager.IotHub/src/autorest.md

```diff
@@ -16,14 +16,84 @@ skip-csproj: true
 modelerfour:
   flatten-payloads: false
 
- 
+override-operation-name:
+  IotHubResource_CheckNameAvailability: CheckIotHubNameAvailability
+  ResourceProviderCommon_GetSubscriptionQuota: GetIotHubUserSubscriptionQuota
+
+rename-mapping:
+  IotHubNameAvailabilityInfo: IotHubNameAvailabilityResponse
+  IotHubNameAvailabilityInfo.nameAvailable: IsNameAvailable
+  OperationInputs: IotHubNameAvailabilityContent
+  Capabilities: IotHubCapability
+  CertificateProperties.created: CreatedOn
+  CertificateProperties.updated: UpdatedOn
+  CertificateProperties.expiry: ExpiryOn
+  CertificatePropertiesWithNonce.created: CreatedOn
+  CertificatePropertiesWithNonce.updated: UpdatedOn
+  CertificatePropertiesWithNonce.expiry: ExpiryOn
+  CertificateVerificationDescription: IotHubCertificateVerificationContent
+  GroupIdInformation: IotHubPrivateEndpointGroupInformation
+  GroupIdInformationProperties: IotHubPrivateEndpointGroupInformationProperties
+  DefaultAction: IotHubNetworkRuleSetDefaultAction
+  AccessRights: IotHubSharedAccessRight
+  FeedbackProperties: CloudToDeviceFeedbackQueueProperties
+  Name: IotHubTypeName
+  EventHubProperties: EventHubCompatibleEndpointProperties
+  RouteProperties: RoutingRuleProperties
+  JobResponse.startTimeUtc: StartOn
+  JobResponse.endTimeUtc: EndOn
+  JobResponse: IotHubJobInfo
+  JobResponseListResult : IotHubJobInfoListResult
+  EndpointHealthData: IotHubEndpointHealthInfo
+  EndpointHealthDataListResult: IotHubEndpointHealthInfoListResult
+  UserSubscriptionQuota.id: IotHubTypeId
+
+prepend-rp-prefix:
+  - AuthenticationType
+  - TestAllRoutesResult
+  - TestAllRoutesInput
+  - TestRouteInput
+  - TestRouteResult
+  - TestRouteResultDetails
+  - TestResultStatus
+  - CertificateProperties
+  - CertificatePropertiesWithNonce
+  - CertificateDescription
+  - CertificateListDescription
+  - CertificateWithNonceDescription
+  - EndpointHealthStatus
+  - EnrichmentProperties
+  - FailoverInput
+  - FallbackRouteProperties
+  - ImportDevicesRequest
+  - PublicNetworkAccess
+  - UserSubscriptionQuota
+  - UserSubscriptionQuotaListResult
+  - IPFilterRule
+  - IPFilterActionType
+  - RoutingSource
+  - JobStatus
+  - JobType
+  - PrivateLinkResources
+  - PrivateEndpointConnectionsList
+  - PrivateLinkServiceConnectionStatus
+  - PrivateEndpointConnectionProperties
+  - RegistryStatistics
+  - MatchedRoute
+  - NetworkRuleSetProperties
+  - NetworkRuleSetIPRule
+  - NetworkRuleIPAction
+  - RoutingProperties
+  - StorageEndpointProperties
 
 format-by-name-rules:
   'tenantId': 'uuid'
   'ETag': 'etag'
   'location': 'azure-location'
   '*Uri': 'Uri'
   '*Uris': 'Uri'
+  'thumbprint': 'any'
```

Comment:
Should we change all the thumbprint to `BinaryData`? I have this question because I noticed that ApiManagement also has this property.

Reply:
most of thime thumbprint is a binary data